pkg/kudoctl/cmd/upgrade.go

```
@@ -0,0 +1,147 @@
+		return fmt.Errorf("please use --instance and specify instance name. It cannot be empty")
+	}
+
+	return nil
+}
+
+func runUpgrade(args []string, options *options) error {
+	err := validate(args, options)
+	if err != nil {
+		return err
+	}
+	packageToUpgrade := args[0]
+
+	kc, err := kudo.NewClient(options.Namespace, viper.GetString("kubeconfig"))
+	if err != nil {
+		return errors.Wrap(err, "creating kudo client")
+	}
+
+	// Resolve the package to upgrade to
+	repository, err := repo.NewOperatorRepository(repo.Default)
+	if err != nil {
+		return errors.WithMessage(err, "could not build operator repository")
+	}
+	crds, err := install.GetPackageCRDs(packageToUpgrade, options.PackageVersion, repository)
+	if err != nil {
+		return errors.Wrapf(err, "failed to resolve package CRDs for operator: %s", packageToUpgrade)
+	}
+
+	return upgrade(crds.OperatorVersion, kc, options)
+}
+
+func upgrade(newOv *v1alpha1.OperatorVersion, kc *kudo.Client, options *options) error {
+	operatorName := newOv.Spec.Operator.Name
+	nextOperatorVersion := newOv.Spec.Version
+
+	// Make sure the instance you want to upgrade exists
+	instance, err := kc.GetInstance(options.InstanceName, options.Namespace)
+	if err != nil {
+		return errors.Wrapf(err, "verifying the instance does not already exist")
+	}
+	if instance == nil {
+		return fmt.Errorf("instance %s in namespace %s does not exist in the cluster", options.InstanceName, options.Namespace)
+	}
+
+	// Check OperatorVersion and if upgraded version is higher than current version
+	ov, err := kc.GetOperatorVersion(instance.Spec.OperatorVersion.Name, options.Namespace)
+	if err != nil {
+		return errors.Wrap(err, "retrieving existing operator version")
+	}
+	if ov == nil {
+		return fmt.Errorf("no operator version for this operator installed yet for %s in namespace %s. Please use install command if you want to install new operator into cluster", operatorName, options.Namespace)
+	}
+	oldVersion, _ := semver.NewVersion(ov.Spec.Version)
```

Should we be suppressing this error?

---

no :)